### ISSUE TITLE
[Accessibility]: cannot navigate out of Monaco with the keyboard

### ISSUE THREAD

**Issue by LawnGnome** [sourcegraph/sourcegraph#36327]
### Audit type

Keyboard navigation

### User journey audit issue

https://github.com/sourcegraph/sourcegraph/issues/34090

### Problem description

Once the Monaco editor on the code host editor page is highlighted, there doesn't appear to be a way to navigate out using only the keyboard.

### Expected behavior

Some method should exist to navigate out.

### Assigning labels

- [ ] Please give this issue an estimate by applying a label like `estimate/Xd`, where X is the estimated number of days it will take to complete.
- [ ] If this issue is specific to a specific Sourcegraph product, please assign the appropriate team label to this issue.
- [ ] If this issue will require input from designers in order to complete, please assign the label `needs-design`.
- [ ] If you are confident that this issue should be fixed by GitStart, please assign the label `gitstart`.

### Owner

I'm unsure who should work on this issue or if it needs more clarification. I will let the Frontend Platform team triage this.

**Comment by umpox** [comment]
This is intentional from Monaco editor and has a frustratingly hard to find shortcut (Ctrl + M) to escape the focus lock. Also reported here: https://github.com/microsoft/monaco-editor/issues/465

Will assign to myself and see if there's anything else we can do here

**Comment by fkling** [comment]
I think we overwrite Tab key behavior in the search query input to navigate out of it. CodeMirror actually intentionally doesn't "trap" Tab by default and even has a section in the documentation about it: https://codemirror.net/examples/tab/

So this might not be an issue anymore if we switch to CodeMirror completely, though given the complex setup for the batch changes and settings editors I don't know how soon that would be.

**Comment by LawnGnome** [comment]
It's a tough one, because I can also see tab being a requirement to really use the Batch Changes editor, given YAML. Could we somehow surface ^M in the UI for keyboard users?

**Comment by felixfbecker** [comment]
Quoting the criterion:

> Success Criterion [2.1.2 No Keyboard Trap](https://www.w3.org/TR/WCAG21/#no-keyboard-trap) (Level A): If keyboard focus can be moved to a component of the page using a [keyboard interface](https://www.w3.org/WAI/WCAG21/Understanding/no-keyboard-trap.html#dfn-keyboard-interface), then focus can be moved away from that component using only a keyboard interface, and, if it requires more than unmodified arrow or tab keys or other standard exit methods, the user is advised of the method for moving focus away.

**Comment by fkling** [comment]
>  the user is advised of the method for moving focus away.

How is that supposed to happen? Is there a recommend way for doing that?

**Comment by felixfbecker** [comment]
It could be with a little muted helper text before the editor (e.g. top right of it) that says "Press <kbd>^</kbd><kbd>M</kbd> to unfocus editor".

It could also be done with `sr-only` text or in `aria-label` of the editor, but it's generally better to also show it to non-screen-reader users. Duplicating it into the `aria-label` can be a good idea though in case the user jumps to it directly.

**Comment by limitedmage** [comment]
I think this should be part of the `aria-label` of the editor, so if you're trapped in it, you can find out how to get out. Would also be good to have it in a visible part of the UI for sighted keyboard users.
